document/CoreFormat.mdk

```diff
@@ -571,6 +571,81 @@ This might happen, for example, if application code for which location informati
 for which location information is not available, which in turn calls back into application code.
 ~
 
+### `relatedLocations` property { #result-relatedLocations }
+
+A `result` object may contain a property named `relatedLocations` whose value is an array
+of one or more `annotatedCodeLocation` objects (&sect;[#annotatedCodeLocation]),
+each of which represents a location relevant to understanding the result.
+
+~ IsoExample
+Suppose that a tool for analyzing JavaScript has a rule that reports a problem
+when a variable declared in an inner scope hides a variable with the same name in an enclosing scope.
+The tool would report the problem on the line where the inner variable is declared.
+The tool could choose to add an element to the `relatedLocations` array, specifying
+the location where the outer variable was declared.
+
+The result might appear in the log file like this:
+
+```
+results: [
+    {
+        "ruleId": "JS3056",
+        "kind": "error",
+        "shortMessage": "Name 'index' cannot be used in this scope because it would give a different meaning to 'index'.",
+
+        "locations": [                  # An array of location objects (see \(&sect;[#location]\))
+            {                           # A location object
+
+                "physicalLocations": [  # An array of physicalLocation objects ...
+                    [                   # ... each of which is an array of physicalLocationComponent objects (see \(&sect;[#physicalLocationComponent]\))
+
+                        {               # A physicalLocationComponent object.
+                            "analysisTarget": "file://C:/Code/a.js",
+                            "region": {
+                                "startLine": "2",
+                                "startColumn": "6"
+                            }
+                        }
+
+                    ]
+                ]
+            }
+        ],
+
+        "relatedLocations": [           # An array of annotatedCodeLocation objects (see \(&sect;[#annotatedCodeLocation]\))
+            {                           # An annotatedCodeLocation object 
+
+                "message": "The previous declaration of 'index' was here.",
+
+                "physicalLocations": [
+                    [
+
+                        {
+                            "analysisTarget": "file://C:/Code/a.js",
+                            "region": {
+                                "startLine": "6",
+                                "startColumn": "10"
+                            }
+                        }
+
+                    ]
+                ]
+            }
+        ]
+    },
+    ...
+]
+```
+
+The tool might write messages to the console like this:
+
+```
+C:\Code\a.js(6,10-10) : error : JS3056: Name 'index' cannot be used in this scope because it would give a different meaning to 'index'.
+C:\Code\a.js(2,6-6) : info : The previous declaration of 'index' was here.
```

Comment:
@michaelcfanning In the example you sent me, the "info" message had its own message id, `JS3116`. Since this information is now part of the result (rather than being a "related result", which we considered as an alternative), IMO it should not have its own message id.

If you think it would clarify, the info message could recap the error id:

```
C:\Code\a.js(2,6-6) : info : JS3056: The previous declaration of 'index' was here.
```